### PR TITLE
Updated hystrix version. Copying MDC Context from parent thread

### PR DESCRIPTION
**Pull request by r0goyal**

### CODE REVIEW

File: src/main/java/io/appform/core/hystrix/GenericHystrixCommand.java [hunk 1]
```diff
@@ -15,37 +15,45 @@
...
 
     private final String traceId;
 
-    public GenericHystrixCommand(HystrixCommand.Setter setter,String traceId) {
+    private final Map parentThreadContext;
+
+    public GenericHystrixCommand(HystrixCommand.Setter setter, String traceId) {
         this.setter = setter;
         this.traceId = traceId;
+        parentThreadContext = MDC.getCopyOfContextMap();
```
Comment (santanusinha) [inline src/main/java/io/appform/core/hystrix/GenericHystrixCommand.java]: Context can be pulled just before CHECK HERE

File: src/main/java/io/appform/core/hystrix/GenericHystrixCommand.java [hunk 2]
```diff
@@ -15,37 +15,45 @@
...
     }
 
     public HystrixCommand<ReturnType> executor(HandlerAdapter<ReturnType> function) throws Exception {
         return new HystrixCommand<ReturnType>(setter) {
```
Comment (santanusinha) [inline src/main/java/io/appform/core/hystrix/GenericHystrixCommand.java:42]: CHECK HERE

File: src/test/java/io/appform/core/hystrix/CommandFactoryTest.java [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 import com.hystrix.configurator.config.HystrixConfig;
 import com.hystrix.configurator.core.HystrixConfigurationFactory;
+import com.netflix.hystrix.HystrixCommand;
+import com.netflix.hystrix.HystrixCommandGroupKey;
+import com.netflix.hystrix.HystrixCommandKey;
```
Comment (santanusinha) [inline src/test/java/io/appform/core/hystrix/CommandFactoryTest.java]: Remove

**Review by santanusinha — CHANGES_REQUESTED** [review]

**Review by santanusinha — APPROVED** [review]

File: src/main/java/io/appform/core/hystrix/GenericHystrixCommand.java [hunk 4]
```diff
@@ -15,37 +15,45 @@
...
             @Override
             protected ReturnType run() throws Exception {
                 try {
+                    if (parentThreadContext != null){
+                        MDC.setContextMap(parentThreadContext);
+                    }
                     MDC.put(TRACE_ID, traceId);
                     return function.run();
-                }finally {
-                    MDC.remove(TRACE_ID);
```
Comment (Thrigunesh) [inline src/main/java/io/appform/core/hystrix/GenericHystrixCommand.java:45]: @r0goyal Any specific reason to clear mdc?

**Pull request merged**